holoviews/plotting/plot.py

```diff
@@ -42,6 +44,45 @@ class Plot(param.Parameterized):
     # Use this list to disable any invalid style options
     _disabled_opts = []
 
+    @property
+    def state(self):
+        """
+        The plotting state that gets updated via the update method and
+        used by the renderer to generate output.
+        """
+        raise NotImplementedError
+
+    @property
+    def root(self):
+        if self._root:
+            return self._root
+        elif 'plot' in self.handles and self.top_level:
+            return self.state
+        else:
+            return None
+
+    @property
+    def document(self):
+        return self._document
+
+    @document.setter
+    def document(self, doc):
+        if (doc and hasattr(doc, 'on_session_destroyed') and
+            self.root is self.handles.get('plot') and not hasattr(self, 'layout_dict')):
```

Comment:
```
not hasattr(self, 'layout_dict')
```

Should fix this condition, it's really looking for AdjointLayoutPlots

Reply:
At least please document that that's what it's doing.